src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -272,15 +276,15 @@ export const useSearch = (services: DiscoverViewServices) => {
       }
       let errorBody;
       try {
-        errorBody = JSON.parse(error.message);
+        errorBody = JSON.parse(error.body);
       } catch (e) {
-        errorBody = error.message;
+        errorBody = error.body;
```

Comment:
how do you know that there is `error.body`? and does it follow type correctly? https://github.com/opensearch-project/OpenSearch-Dashboards/blob/7da544372cd24c80eaf286a143d4c844c6087c60/src/plugins/discover/public/application/view_components/utils/use_search.ts#L55-L59